contracts/pre-propose/dao-pre-propose-approval-single/src/contract.rs

```diff
@@ -406,7 +414,129 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
+
+            let pending_proposals = dao_pre_propose_approval_single_v241::state::PENDING_PROPOSALS
+                .range(deps.storage, None, None, Order::Ascending)
+                .collect::<StdResult<Vec<_>>>()?;
+            for (id, proposal) in pending_proposals {
+                PENDING_PROPOSALS.save(
+                    deps.storage,
+                    id,
+                    &Proposal {
+                        status: ApprovalProposalStatus::Pending {},
+                        approval_id: proposal.approval_id,
+                        approver: approver.clone(),
+                        proposer: proposal.proposer,
+                        msg: ProposeMsg {
+                            title: proposal.msg.title,
+                            description: proposal.msg.description,
+                            msgs: proposal.msg.msgs,
+                            proposer: proposal.msg.proposer,
+                            vote: proposal.msg.vote.map(|vote| SingleChoiceAutoVote {
+                                vote: match vote.vote {
+                                    dao_voting_v241::voting::Vote::Yes => Vote::Yes,
+                                    dao_voting_v241::voting::Vote::No => Vote::No,
+                                    dao_voting_v241::voting::Vote::Abstain => Vote::Abstain,
+                                },
+                                rationale: vote.rationale,
+                            }),
+                        },
+                        deposit: proposal.deposit.map(|deposit| CheckedDepositInfo {
+                            denom: match deposit.denom {
+                                cw_denom_v241::CheckedDenom::Native(denom) => {
+                                    CheckedDenom::Native(denom)
+                                }
+                                cw_denom_v241::CheckedDenom::Cw20(addr) => CheckedDenom::Cw20(addr),
+                            },
+                            amount: deposit.amount,
+                            refund_policy: match deposit.refund_policy {
+                                dao_voting_v241::deposit::DepositRefundPolicy::Always => {
+                                    DepositRefundPolicy::Always
+                                }
+                                dao_voting_v241::deposit::DepositRefundPolicy::OnlyPassed => {
+                                    DepositRefundPolicy::OnlyPassed
+                                }
+                                dao_voting_v241::deposit::DepositRefundPolicy::Never => {
+                                    DepositRefundPolicy::Never
+                                }
+                            },
+                        }),
+                    },
+                )?;
+            }
+
+            let completed_proposals =
+                dao_pre_propose_approval_single_v241::state::COMPLETED_PROPOSALS
+                    .range(deps.storage, None, None, Order::Ascending)
```

Comment:
wonder from how many props this could fail. maybe worth to do some test runs?

Reply:
yeah.. what's the easiest way to test this?

Reply:
both test-tube and interchaintest provide this. probably interchaintest would be easiest to test across multiple chains, but require a bit more upfront setup than test-tube

Reply:
alternatively, we could make the approver field optional, and if it's empty, default to the currently configured approver! that way, it's backwards compatible and requires no migration 🙌

Reply:
> alternatively, we could make the approver field optional, and if it's empty, default to the currently configured approver! that way, it's backwards compatible and requires no migration 🙌

actually nevermind. if any of the types change serialization in the future, not having a migration flow that force-serializes with new types can lead to unparseable state. dang.